src/runner/browser-job.js

```diff
@@ -99,7 +102,7 @@ export default class BrowserJob extends EventEmitter {
 
     // API
     get hasQueuedTestRuns () {
-        return !!this.testRunQueue.length;
+        return !!this.testRunQueue.length || this.opts.quarantineMode && !this.done;
```

Comment:
With this approach we will always have on job left in the queue after test run. However, I agree with @AlexanderMoskovkin points. We need something conceptually better.